### PR TITLE
refactor(log): support enum log level

### PR DESCRIPTION
**Pull request by cdaringe**
# problem

- on setting up my own logger instance, i observed that exported log levels by levelName were of type `Record<string, number>`, and did not constrain my input to a valid, case-sensitive, stringy value

# solution

- `enum`ify  and expose `LogLevels`
- type `levelName` to have type `LogLevel`, which is just a good ol' string union of `"INFO" | "DEBUG" | ...` etc
- update `mod.ts` with enum export
- update docs w/ inline examples _using_ said enum export
- increase strictness around providing bogus log levels

# discussion

this is one of those topics that i think people can validly debate if more typing helps or hurts. i'm not married to this proposal or the implementation details, so by all means--fire away. i _do_ want type safety here and am fine to bring it into userspace if we decide we dont want such _stuff_ here.

### CODE REVIEW

**Comment by ry** [comment]
Thanks for the patch - the CI is failing on lint and/or formatting. Please run `./tools/format.py --js` and `./tools/lint.py --js`

File: std/log/levels.ts [hunk 1]
```diff
@@ -1,26 +1,51 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
-export const LogLevel: Record<string, number> = {
-  NOTSET: 0,
-  DEBUG: 10,
-  INFO: 20,
-  WARNING: 30,
-  ERROR: 40,
-  CRITICAL: 50,
-};
+export enum LogLevels {
```
Comment (bartlomieju) [inline std/log/levels.ts:2]: `LogLevels` and `LogLevel` are too similar to each other, I wasn't sure what was the change for a few minutes. Could we use some more distinct names? Also, it'd be great if you could add JSdocs explaining purpose of each export

Comment (cdaringe) [reply std/log/levels.ts:2]: JSDoc--sure, no problem. I found the nearness of the symbols to be a _strength_--that proximity infers their tight coupling and consolidates your import lookup keystrokes. With that said, if you still feel strongly about it, i'm happy to take a prescription and just type them in. `LogLevel` to just `Level` or `LevelName`, perhaps?

Comment (cdaringe) [reply std/log/levels.ts:4]: `LevelName` seems to bode well, as it also has parity with `levelName` args in various existing functions. tell me how you like e6d508b

**Pull request merged**